app/features/receive/use-receive-cashu-token.tsx

```diff
@@ -0,0 +1,358 @@
+import { CashuMint, CashuWallet, type Token } from '@cashu/cashu-ts';
+import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+import type { Account, CashuAccount } from '~/features/accounts/account';
+import {
+  useAccounts,
+  useAddAccount,
+  useDefaultAccount,
+} from '~/features/accounts/account-hooks';
+import { tokenToMoney } from '~/features/shared/cashu';
+import { useToast } from '~/hooks/use-toast';
+import {
+  getP2PKPubkeyFromProofs,
+  getUnspentProofsFromToken,
+  isP2PKSecret,
+  isPlainSecret,
+  swapProofsToWallet,
+} from '~/lib/cashu';
+import { checkIsTestMint, getMintInfo } from '~/lib/cashu';
+import type { MintInfo } from '~/lib/cashu';
+import type { Money } from '~/lib/money';
+import { useNavigateWithViewTransition } from '~/lib/transitions';
+import { getDefaultUnit } from '../shared/currencies';
+
+type UseReceiveCashuTokenProps = {
+  token: Token;
+  cashuPubKey: string;
+};
+
+type UseReceiveCashuTokenData = {
+  /** The token with only claimable proofs or the full token if it cannot be claimed */
+  displayToken: Token;
+  /** Whether the token can be claimed (fully or partially) by this user */
+  canClaim: boolean;
+  /** The account to receive the token */
+  receiveAccount: CashuAccount;
+  /** The account that the token is from */
+  sourceAccount: CashuAccount;
+  /** True if the source account cannot make a lightning payment to other accounts */
+  disableCrossMintSwap: boolean;
+  /** Whether the selected account is the source mint */
+  selectedAccountIsSource: boolean;
+  /** The reason why the user cannot claim the token */
+  cannotClaimReason: string | undefined;
```

Comment:
should we maybe change the type to be union type that will have `cannotClaimReason` as string only when canClaim is false?